internal/contour/cachehandler.go

```diff
@@ -47,6 +52,8 @@ func (ch *CacheHandler) OnChange(dag *dag.DAG) {
 	ch.updateRoutes(dag)
 	ch.updateClusters(dag)
 
+	ch.SnapshotHandler.UpdateSnapshot()
```

Comment:
are the `internal/contour` caches even necessary anymore? It seems like in the `updateSecrets`, etc. methods here, we could just return the resources directly, and then pass them to `UpdateSnapshot`. I might be missing something about how the contour caches are used, though, so take my comment with a grain of salt :)

Reply:
No I don't think they are, we could just pick them out of the dag rebuild. Endpoints currently are the only caches left that would need to stay since they are managed per change.

Reply:
Thanks for the suggestion @skriss, I've done some refactoring on this.